src/pyhf/__init__.py

```diff
@@ -28,7 +28,7 @@ def get_backend():
 
 
 @events.register('change_backend')
-def set_backend(backend, custom_optimizer=None):
+def set_backend(backend, custom_optimizer=None, mode="GPU"):
```

Comment:
```suggestion
def set_backend(backend, custom_optimizer=None, mode="CPU"):
```

I think CPU should be the default. Additionally, we should allow for lowercase `cpu` and `gpu` too. The easiest way to accommodate this is to do something like

```python
if mode not in ['cpu', 'gpu', 'CPU', 'GPU']:
  raise ValueError(f"The mode must be either 'cpu' or 'gpu', not {mode}.")
tensorlib.mode = mode.lower()
```

Reply:
Note that we should only allow `mode` to be used if the backend specified is a string -- because we'll be instantiating the backend for the user. Otherwise, we shouldn't do anything if the user passes in a custom backend.